packages/commands/src/components/command-menu.js

```diff
@@ -90,17 +90,31 @@ export function CommandMenuLoaderWrapper( { hook, search, setLoader, close } ) {
 }
 
 export function CommandMenuGroup( { group, search, setLoader, close } ) {
+	const hasSearch = !! search;
 	const { commands, loaders } = useSelect(
 		( select ) => {
-			const { getCommands, getCommandLoaders } = select( commandsStore );
+			const {
+				getCommands,
+				getCommandLoaders,
+				getContextualCommands,
+				getContextualCommandLoaders,
+			} = select( commandsStore );
 			return {
-				commands: getCommands( group ),
-				loaders: getCommandLoaders( group ),
+				commands: hasSearch
+					? getCommands( group )
+					: getContextualCommands( group ),
+				loaders: hasSearch
+					? getCommandLoaders( group )
+					: getContextualCommandLoaders( group ),
 			};
 		},
-		[ group ]
+		[ group, hasSearch ]
 	);
 
+	if ( ! commands.length && ! loaders.length ) {
```

Comment:
Here we can have loaders but not commands. You can test this by invoking it in a theme template without changes.

Reply:
That's normal, but ideally it shouldn't show the empty space it's showing now.

Reply:
Solved